### PR TITLE
Read Xmp.exif.DateTimeOriginal from xmp sidecar

### PR DESCRIPTION
**Pull request by mueller-physics**
This allows to override the date and time a picture was taken by adding
a time-stamp to the sidecar file, via

exiv2 -M"set Xmp.exif.DateTimeOriginal 2017:10:23 12:34:56" rawfile.ext.xmp
or
exiftool '-datetimeoriginal=2017:10:23 12:34:56' rawfile.ext.xmp

This helps to fix photos with broken time-stamps (clock set wrong in camera, scanned photos, etc.) w/o modifying the raw files.

### CODE REVIEW

**Comment by LebedevRI** [comment]
Also see https://redmine.darktable.org/issues/10569 (Not about this tag specifically)
> TL DR: yes, the fix is right, but it would break usecase of single XMP for multiple different(not duplicated) images, so until next big metadata handling overhaul, it will have to stay as it is.

TLDR: can't be fixed.

**Comment by mueller-physics** [comment]
I have seen the issue, but assumed the problem was only actively writing tags to the sidecar, not reading in existing ones (that would have been put into the side-car on purpose)? At least, other tags (Xmp.exif.GPS..., Xmp.exifEX.LensModel) already override EXIF data if darktable finds them in the XMP file.

To fix this, could there be a switch somewhere in config, to enable a "one side-car file per image" use-case? Or any other way to reliable store "corrections" to meta-data w/o touching the original files or relying on the darktable database (which is much harder to backup)?

**Comment by LebedevRI** [comment]
In 
> usecase of single XMP for multiple different(not duplicated) images

i of course was talking not about dt proper, but dt-cli, where you can specify the sidecar to be used.

**Comment by mueller-physics** [comment]
Ah, thanks, I was wondering if I missed some case in the GUI where duplicated files could share a side-car. But for the dt-cli use-case:
- why would someone reuse an xmp that has overrides in it (that have to be added manually, currently on the command line) if they do not indent to apply these overrides
- why not offer a switch to ignore those overrides when using dt-cli (that might default to 'ignore')?

Thanks for your help, by the way, great community!

**Comment by houz** [comment]
I think this PR makes sense and should be merged. dt itself doesn't write the time stamp to the sidecar, so it's the user's responsibility to deal with that.
